### DISCUSSION TITLE
Can I use EnvironmentStateObject with the @Observable macro?

### DISCUSSION THREAD

**Discussion by GregoryZeng** [groue/GRDBQuery#55] [Q&A]
Hi,
I am a beginner in SwiftUI and GRDB so correct me if I am wrong 😂
The view models from the docs and examples use the Combine-style `ObservableObject`. 
https://swiftpackageindex.com/groue/grdbquery/0.9.0/documentation/grdbquery/mvvm
https://github.com/groue/GRDBQuery/tree/main/Documentation/MVVMDemo

I wonder if the `@Observable` macro can also apply to the view model with GRDB, removing the `@Published` and the `ObservableObject`? 
https://developer.apple.com/documentation/swiftui/migrating-from-the-observable-object-protocol-to-the-observable-macro

Thanks.

**Comment by groue** [comment]
Hello, @GregoryZeng,

`ObservableObject` requires an `ObservableObject`, and is not able to deal with observable classes decorated with `@Observable`.

I don't have any ready-made option for instantiating `@Observable` instances from the environment.

In your own explorations, make sure you carefully read the [Store observable objects](https://developer.apple.com/documentation/swiftui/state#Store-observable-objects) chapter of the `@State` documentation.

**Comment by GregoryZeng** [comment]
Thanks for your prompt answer.

> ObservableObject requires an ObservableObject

You mean `EnvironmentStateObject` requires an `ObservableObject`?

OK, I thought that `ObservableObject` can be [migrated](https://developer.apple.com/documentation/swiftui/migrating-from-the-observable-object-protocol-to-the-observable-macro#Migrate-incrementally) to `@Observable` directly ...

**Reply by groue** [reply]
Yes, thank you for fixing my typo :-)
